src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1549,6 +1550,32 @@ export class SearchView extends ViewletPanel {
 		}, errors.onUnexpectedError);
 	}
 
+	openEditorWithMultiCursor(element: FileMatchOrMatch): Promise<void> {
+		const resource = element instanceof Match ? element.parent().resource : (<FileMatch>element).resource;
+		return this.editorService.openEditor({
+			resource: resource,
+			options: {
+				preserveFocus: false,
+				pinned: true,
+				revealIfVisible: true
+			}
+		}).then(editor => {
+			if (editor) {
+				let codeEditor = getCodeEditor(editor.getControl());
+				if (codeEditor) {
+					let multiCursorController = MultiCursorSelectionController.get(codeEditor);
+					let isRegex = this.searchWidget.searchInput.getRegex();
+					let isWholeWords = this.searchWidget.searchInput.getWholeWords();
+					let isCaseSensitive = this.searchWidget.searchInput.getCaseSensitive();
+					let contentPattern = this.searchWidget.searchInput.getValue();
```

Comment:
This is coming from the search input, not the query, so the user could have changed this without rerunning the search, then it won't be the same, right?

Actually, can we avoid triggering a find, and just select the exact ranges that are in the actual match?

Reply:
Two questions:

1. How do I get all the matches in `searchView.ts`?
2. Should I create a public wrapper method for [_setSelections](https://github.com/microsoft/vscode/blob/27eafea04aaf47b1434efbd9c74c2f67391044bd/src/vs/editor/contrib/multicursor/multicursor.ts#L498) method and use that in `searchView.ts` or this task need to be done in some other way?

Reply:
You have the matches right here in the FileMatchOrMatch

Reply:
Done @roblourens